config_system/config_system/general.py

```diff
@@ -491,24 +491,23 @@ def set_config(key, value, is_user_set=True):
         logger.warning("Ignoring unknown configuration option %s" % key)
         return
 
-    if is_user_set:
-        # Validate user input
-        if c['datatype'] == 'bool':
-            # Must be y or n
-            if value not in [True, False]:
-                logger.warning(
-                    "Ignoring boolean configuration option %s with non-boolean value '%s'." % (
-                        key, value))
-                return
-        elif c['datatype'] == 'int':
-            # Must convert to an integer
-            try:
-                value = int(value)
-            except ValueError:
-                logger.warning(
-                    "Ignoring integer configuration option %s with non-integer value '%s'" % (
-                        key, value))
-                return
+    # Validate input
```

Comment:
Cool - so this now validates all options, not just user_set ones. Makes sense, and I'm surprised we weren't doing this already.